lib/pbench/test/unit/server/test_unpack_tarballs.py

```diff
@@ -0,0 +1,421 @@
+from collections import namedtuple
+import os
+from pathlib import Path
+from typing import Any
+
+import pytest
+
+from pbench.common.logger import get_pbench_logger
+from pbench.server.filetree import FileTree, TarballUnpackError
+from pbench.server.unpack_tarballs import UnpackTarballs
+
+
+@pytest.fixture()
+def make_logger(server_config):
+    """
+    Construct a Pbench Logger object
+    """
+    return get_pbench_logger("TEST", server_config)
+
+
+class TestUnpackTarballs:
+    tarlist = ["/mock/ABC/TO-UNPACK/A.tar.xz", "/mock/ABC/TO-UNPACK/B.tar.xz"]
+    tar = Path(tarlist[0])
+    min_size, max_size, tar_size, tar_mode = 0, 1000, 586, 16893
+
+    def mock_stat(self, *args, follow_symlinks=False):
+        """Mocked replacement for Path.stat()"""
+        StatResult = namedtuple("StatResult", ["st_mode", "st_size"])
+        return StatResult(
+            st_mode=TestUnpackTarballs.tar_mode, st_size=TestUnpackTarballs.tar_size
+        )
```

Comment:
Are you [still considering](https://github.com/distributed-system-analysis/pbench/pull/2975#discussion_r985627161) having this mock return different sizes for the different tarballs?